pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -313,11 +324,14 @@ func (c *Controller) updateOperatorStatus(ctx context.Context, previousStatus *o
 		c.versionRecorder.SetVersion(operandName, c.targetOperandVersion)
 	}
 
-	// set updateGenerationFn so that it is invoked in defer
-	updateGenerationFn = func(newStatus *operatorv1.OperatorStatus) error {
-		resourcemerge.SetDeploymentGeneration(&newStatus.Generations, workload)
-		return nil
-	}
+	// set generations
+	status = status.WithGenerations(&applyoperatorv1.GenerationStatusApplyConfiguration{
```

Comment:
need to ensure *every* branch has generations set.  Probably add a check in that defer.

Reply:
Created the status with the generations filled (like we do in the deployment controller).